pages/desktop/user.py

```diff
@@ -119,8 +124,28 @@ def click_update_account(self):
         self.selenium.find_element(*self._update_account_locator).click()
         WebDriverWait(self.selenium, self.timeout).until(lambda s: self.update_message == "Profile Updated")
 
-    def change_hide_email_state(self):
-        self.selenium.find_element(*self._hide_email_checkbox).click()
+    def hide_email_address(self):
+        email_checkbox = self.selenium.find_element(*self._email_checkbox)
+        WebDriverWait(self.selenium, self.timeout).until(expected_conditions.visibility_of(email_checkbox));
+
+        if email_checkbox.is_selected():
+            err_msg = 'Email is already hidden'
+            return err_msg
+
+        if not email_checkbox.is_selected():
```

Comment:
This `if` statement is not needed as the code will never reach this point unless the condition is true. I.e., if the email_checkbox was selected we would have already thrown the above error (or returned the error message, in your current code).